### ISSUE TITLE
⌨ Hide "Mark all as read" when notifications are read or empty

### ISSUE THREAD

**Issue by yonikid15** [GCTC-NTGC/gc-digital-talent#10646]
## ⌨ Accessibility Issue

The "Mark all as read" button serves no purpose when no notifications are present. Only show "Mark all as read" button when there is at least one or more unread notification(s). It's also making a silent request in the background, which is redundant.

### WCAG 2.1 Criteria

List any relevant criteria that is failing.

## 🦋 Expected Behaviour

Button should only be visible when there are unread notifications.

## 📋 Steps to Reproduce

1. Go to list of notifications with read notifications and still see button is visible.

## 📸 Screenshot

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/22059495/fb38db89-3a06-4dac-8e53-35248d6c2dc1)

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by gobyrne** [comment]
I thought we already debated this one and decided to leave it as is.

**Comment by gobyrne** [comment]
Yeah this was debated in #9962 in the end we decided to just leave the button, even when there are no notifications present.
